### DISCUSSION TITLE
Apex rule priorities seem incorrect?

### DISCUSSION THREAD

**Discussion by jfeingold35** [pmd/pmd#3911] [Q&A]
The priorities assigned to Apex rules don't seem correct to me.
E.g., `ClassNamingConventions` is a code style rule [whose priority is 1 (the highest value)](https://github.com/pmd/pmd/blob/master/pmd-apex/src/main/resources/category/apex/codestyle.xml#L26), as are [`FieldNamingConventions`](https://github.com/pmd/pmd/blob/master/pmd-apex/src/main/resources/category/apex/codestyle.xml#L145), and [`FormalParameterNamingConventions`](https://github.com/pmd/pmd/blob/master/pmd-apex/src/main/resources/category/apex/codestyle.xml#L193), while security rules like [`ApexCRUDViolation`](https://github.com/pmd/pmd/blob/master/pmd-apex/src/main/resources/category/apex/security.xml#L96), [`ApexInsecureEndpoint`](https://github.com/pmd/pmd/blob/master/pmd-apex/src/main/resources/category/apex/security.xml#L159), and [`ApexSharingViolations`](https://github.com/pmd/pmd/blob/master/pmd-apex/src/main/resources/category/apex/security.xml#L205) all have priority 3, which is the lowest value.
Is this intentional? If so, what's the historical reason for this being the case?

**Comment by rsoesemann** [comment]
There was no reason. It most probably was an accident. Please submit a PR and add a real issue to fix that.

**Reply by jfeingold35** [reply]
Is it possible that the severities are related to level of confidence in the rule? Like, maybe the FLS rules are noisy, so their severity is lower, while the rules about names are less likely to be false positives?

**Reply by rsoesemann** [reply]
No, not at all. Most of the rules were converted by me in one big chuck from Java to Apex. I am quite confident that no one ever put thought into this. Maybe use the blame function in Git to see that no one ever changed them ;-)
